### DISCUSSION TITLE
problem getting namedVersions using @itwin/imodels-client-authoring

### DISCUSSION THREAD

**Discussion by spowell-216** [iTwin/itwinjs-core#3493] [Q&A] [answered]
I'm in the process of upgrading to 3.x and attempting to get list of the last 2 named versions for an iModel using the [@itwin/imodels-client-authoring](https://github.com/iTwin/imodels-clients).

I'm getting the following error: `Unauthorized: The user is unauthorized. Please provide valid authentication credentials.`

the decoded token that is being used has the following scopes: 
![image](https://user-images.githubusercontent.com/51784707/162245717-3f860a1e-c45b-43b7-8cbc-789c9eee4d8b.png)

the same service account can create, download, and delete iModels using `IModelHost.hubAccess`

code based upon sample in [readme](https://github.com/iTwin/imodels-clients/blob/main/clients/imodels-client-management/README.md#get-all-imodel-changesets)
``` 
      const token: string = await IModelHost.getAccessToken();
      const iModelsClient: IModelsClient = new IModelsClient();
      
      // get the list of versions
      const verArgs: GetNamedVersionListParams = { 
        authorization: () =>  { return Promise.resolve({ scheme: "Bearer", token: token })}, 
        iModelId: iModelId, 
        urlParams: { $top: 2 } 
      };

      const versions = iModelsClient.namedVersions.getMinimalList(verArgs);
      const verArray: MinimalNamedVersion[] = [] 
      for await(const value of versions) {
        verArray.push(value);
      }
```

packages used: 
```
"dependencies": {
    "@itwin/core-backend": "3.1.1",
    "@itwin/core-bentley": "3.1.1",
    "@itwin/core-common": "3.1.1",
    "@itwin/core-geometry": "3.1.1",
    "@itwin/imodels-access-backend": "^1.0.1",
    "@itwin/imodels-client-authoring": "^1.0.1",
    "@itwin/service-authorization": "^0.6.2"
  },
```

any help would be much appreciated.

**Comment by spowell-216** [comment, marked as answer]
never mind, i figured it out.  The authorization callback function needs to looks something like this when using `IModelHost.getAccessToken()` since the this token returns "Bearer ey..."

```
private async getAuthorization(): Promise<Authorization> { 
    const token: string = await IModelHost.getAccessToken();
    const tokenDeconstruct: string[] =  token.split(" ");
    return { scheme: tokenDeconstruct[0], token: tokenDeconstruct[1] }
}
```

not that obvious when initially looking at the sample in the [readme](https://github.com/iTwin/imodels-clients/blob/main/clients/imodels-client-management/README.md#get-all-project-imodels)

**Reply by calebmshafer** [reply]
@austeja-bentley @karolisab can you guys take a look at updating the README?

**Reply by austeja-bentley** [reply]
sure, will try to update the README by the end of this week. Thanks for bringing this up.
